lib/active_record/connection_adapters/mysql_flexmaster_adapter.rb

```diff
@@ -252,12 +255,14 @@ def initialize_connection(host, port)
               cx.query_options.merge!(:as => :array)
             end
           end
-        rescue Mysql2::Error => e
-          collected_errors << e
-          nil
-        rescue Timeout::Error => e
-          collected_errors << e
-          nil
+        rescue Mysql2::Error, Timeout::Error => e
+          if retries < @connection_retries
+            retries += 1
+            retry
```

Comment:
just to be clear, this *retries* twice, for a total of 3 times?  Sorry to nitpick but tries vs. retries always gets a bit murky, I think just calling it "CONNECTION_ATTEMPTS" is crystal clear and doesn't run into head scratching.